src/components/templates/Search/Filters.tsx

```diff
@@ -23,39 +24,61 @@ const accessFilterItems = [
   { display: 'compute ', value: FilterByAccessOptions.Compute }
 ]
 
+const complianceFilterItems = [
+  { display: 'in compliance', value: FilterByIsInComplianceOptions.True }
+]
+
 export default function FilterPrice({
   serviceType,
   accessType,
+  complianceType,
   setServiceType,
   setAccessType,
+  setComplianceType,
   addFiltersToUrl,
   className
 }: {
   serviceType: string
   accessType: string
+  complianceType: string
   setServiceType: React.Dispatch<React.SetStateAction<string>>
   setAccessType: React.Dispatch<React.SetStateAction<string>>
+  setComplianceType: React.Dispatch<React.SetStateAction<string>>
   addFiltersToUrl?: boolean
   className?: string
 }): ReactElement {
   const navigate = useNavigate()
   const [serviceSelections, setServiceSelections] = useState<string[]>([])
   const [accessSelections, setAccessSelections] = useState<string[]>([])
+  const [complianceSelections, setComplianceSelections] = useState<string[]>([])
 
   async function applyFilter(filter: string, filterType: string) {
-    filterType === 'accessType' ? setAccessType(filter) : setServiceType(filter)
+    if (filterType === 'accessType') {
+      setAccessType(filter)
+    } else if (filterType === 'serviceType') {
+      setServiceType(filter)
+    } else if (filterType === 'complianceType') {
+      setComplianceType(filter)
+    }
+
     if (addFiltersToUrl) {
       let urlLocation = ''
       if (filterType.localeCompare('accessType') === 0) {
         urlLocation = await addExistingParamsToUrl(location, ['accessType'])
-      } else {
+      } else if (filterType.localeCompare('serviceType') === 0) {
         urlLocation = await addExistingParamsToUrl(location, ['serviceType'])
+      } else if (filterType.localeCompare('complianceType') === 0) {
+        urlLocation = await addExistingParamsToUrl(location, ['complianceType'])
       }
 
       if (filter && location.search.indexOf(filterType) === -1) {
-        filterType === 'accessType'
-          ? (urlLocation = `${urlLocation}&accessType=${filter}`)
-          : (urlLocation = `${urlLocation}&serviceType=${filter}`)
+        if (filterType === 'accessType') {
+          urlLocation = `${urlLocation}&accessType=${filter}`
+        } else if (filterType === 'serviceType') {
+          urlLocation = `${urlLocation}&serviceType=${filter}`
+        } else if (filterType === 'complianceType') {
```

Comment:
Hmzz, Not sure this is correct. Compliance can both be true or false for services (not sure what accessType is).

Please create a ticket for this, as I highly doubt this is correct (same applies to the code above for the compliance filter parts)

Reply:
The value of `accessType` can be 'Download' or 'Compute' and it is set during the Dataset (serviceType) publication. Download means public access and Compute means private access, which allows the data to be processed only by selected Algorthms (serviceType). Reference: https://docs.oceanprotocol.com/building-with-ocean/compute-to-data/compute-to-data-datasets-algorithms The filters can be applied together or separately. I'm not sure if I understand what's wrong.

Reply:
In addition Algorithm (serviceType) is always Compute